src/pages/ProductListing.js

```diff
@@ -0,0 +1,192 @@
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import Navbar from "../components/Navbar";
+import { getProductsByCategory } from "../api-calls";
+import { useNotification } from "../contexts/notificationContext";
+import ProductListingcard from "../components/ProductListingCard";
+import { filter } from "../utils/productFilter";
+
+const ProductListing = () => {
+
+    const { category } = useParams();
+    const { notificationHandler } = useNotification();
+    const [products, setProducts] = useState();
+    const [activeFilters, setActiveFilters] = useState([]);
+    const [filteredProducts, setFilteredProducts] = useState();
+
+    useEffect(() => {
+
+        (async () => {
+
+            try {
+                const products = await getProductsByCategory(category);
+                setProducts(products);
+            }
+            catch (error) {
+
+                notificationHandler(error.message);
```

Comment:
suggestion: instead of using iife, you can create a function and call that function here.